sunpy/io/_file_tools.py

```diff
@@ -58,6 +58,42 @@ def __getitem__(self, key):
 })
 
 
+def get_valid_filepath(filepath):
+    """
+    Ensures the filepath is a valid string, pathlib.Path, or fsspec.OpenFile.
```

Comment:
Was the original start to always return a fsspec.OpenFile regardless?

Reply:
I am not exactly sure what you mean by that. As I make of it, fsspec.OpenFile should be returned only if filepath is not OpenFile, else there doesn't seem to be any merit to do so. It should be returned as-is in that case.

Reply:
The original issue had the line:

"Firstly, io._file_tools should support being passed an [fsspec.OpenFile](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.core.OpenFile) object rather than just a path (if passed a path it should convert to an OpenFile)"

See the part at the end.